starter-code/src/functions-and-arrays.js

```diff
@@ -33,68 +70,79 @@ var wordsArr = [
   'palace'
 ];
 
-// Unique Arrays
-var wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
-];
+function averageWordLength(array){
+  if(array.length === 0)
+    return undefined;
 
-// Finding Elements
-var wordsFind = [
-  'machine',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'truth',
-  'disobedience'
-];
+  for (var result = 0, i = 0; i < array.length; i++){
+    result += array[i].length / array.length
+  }
+  result = Math.round(result);  
+  return result;
+}
 
-// Counting Repetion
-var wordsCount = [
-  'machine',
-  'matter',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'matter',
-  'truth',
-  'disobedience',
-  'matter'
-];
-// Bonus Quest
+// // Unique Arrays
```

Comment:
for finding a unique array you can try something like this 
`
function uniquifyArray (array) {
  var newArray = [];

  array.map (function (word) {
    if (newArray.indexOf(word) == -1)
      newArray.push(word);
  });

  return newArray;
}
`